### DISCUSSION TITLE
launchOptions are not work for Cypress 13.3.0

### DISCUSSION THREAD

**Discussion by utkukilincci** [cypress-io/cypress#28030] [Questions and Help]
```
e2e: {
    setupNodeEvents(on, config) {
      on('before:browser:launch', (browser, launchOptions) => {
        if (browser.name === 'chrome' && browser.isHeadless) {
          launchOptions.args.splice(50, 1)
          launchOptions.args.push('--window-size=1920,1080');
        }

        if (browser.name === 'electron' && browser.isHeadless) {
          launchOptions.preferences.width = 1920;
          launchOptions.preferences.height = 1080;
        }

        return launchOptions;
      });
      
 }
```

      
  I have a config like this. When I use Cypress 10.9.0 this configs work. but in cypress v13 don't.  when my tests are failed, screen shots saved as 1920x950. but I want 1920x1080
  
 My tests running headless
  
![Screenshot 2023-10-11 at 17 34 31](https://github.com/cypress-io/cypress/assets/69322896/79c0e6f0-4189-48ed-82ee-e9a1a26aeafd)

**Comment by portyu9** [comment]
Hi @utkukilincci 

`before:browser:launch` changes the **browser's outer window**, but that is not the same thing as setting the Cypress application's viewport. That is why `--window-size=1920,1080` is not a reliable way to guarantee that a failure screenshot itself is exactly 1920×1080.

If what you need is a 1920×1080 AUT viewport, configure Cypress directly:

```js
const { defineConfig } = require('cypress')

module.exports = defineConfig({
  viewportWidth: 1920,
  viewportHeight: 1080,

  e2e: {
    setupNodeEvents(on, config) {
      // other events
    },
  },
})
```

Also note that automatic screenshots on test failure use Cypress's **runner capture**, not simply the AUT viewport. The runner/Command Log and browser geometry can therefore affect the resulting image dimensions.

If you still need to control Chrome's outer headless window, avoid this:

```js
launchOptions.args.splice(50, 1)
```

Argument position `50` is an implementation detail and can change between Cypress/browser versions. Remove an existing `--window-size` argument by value instead:

```js
launchOptions.args = launchOptions.args.filter(
  (arg) => !arg.startsWith('--window-size=')
)

launchOptions.args.push('--window-size=1920,1080')
return launchOptions
```

So use `viewportWidth` / `viewportHeight` for the tested page size, and only use browser launch arguments when you specifically need to control the browser's outer window.

Please mark this answer as accepted if it helped, thank you.
